graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DmlFetcher.java

```diff
@@ -88,6 +99,118 @@ protected DmlFetcher(
     }
   }
 
+  protected List<TypedKeyValue> buildTypedKeyValueList(
+      Table table, DataFetchingEnvironment environment) {
+    return buildTypedKeyValueList(buildClause(table, environment));
+  }
+
+  protected List<TypedKeyValue> buildTypedKeyValueList(List<Relation> relations) {
+    List<TypedKeyValue> typedKeyValues = new ArrayList<>();
+    for (Relation rel : relations) {
+      if (rel instanceof DefaultRelation) {
+        LeftOperand leftOperand = ((DefaultRelation) rel).getLeftOperand();
+        Term term = ((DefaultRelation) rel).getRightOperand();
+
+        boolean queryByEntries = false;
+        CqlIdentifier columnId;
+        if (leftOperand instanceof ColumnComponentLeftOperand) {
+          columnId = ((ColumnComponentLeftOperand) leftOperand).getColumnId();
+          queryByEntries = true;
+        } else {
+          columnId = ((ColumnLeftOperand) leftOperand).getColumnId();
+        }
+
+        Column column = getColumn(table, columnId.asInternal());
+        assert term != null;
+
+        if (isOrContainsUDT(Objects.requireNonNull(column.type()))) {
+          // Null out the value for now since UDTs are not allowed for use with custom authorization
+          typedKeyValues.add(
+              new TypedKeyValue(
+                  column.cqlName(), Objects.requireNonNull(column.type()).cqlDefinition(), null));
+          continue;
+        }
+
+        if (term instanceof TupleTerm) {
```

Comment:
When do we hit that branch? The only place where the driver's query builder creates a `TupleTerm` is `QueryBuilder.tuple()`, and it's not referenced anywhere in this project. `DmlFetcher.toCqlTerm` only creates instances of `RawTerm`.

Reply:
I ran into this for "IN" queries like

```sql
SELECT l FROM collections."PkListTable" WHERE l IN ([1,2,3],[4,5,6])
```

Reply:
Ah my bad, that's indeed how IN relations work.